src/openrct2/drawing/Line.cpp

```diff
@@ -16,44 +16,43 @@
  * Draws a horizontal line of specified colour to a buffer.
  *  rct2: 0x0068474C
  */
-static void gfx_draw_line_on_buffer(rct_drawpixelinfo* dpi, char colour, int32_t y, int32_t x, int32_t no_pixels)
+static void gfx_draw_line_on_buffer(rct_drawpixelinfo* dpi, char colour, const ScreenCoordsXY& coords, int32_t no_pixels)
 {
-    y -= dpi->y;
+    ScreenCoordsXY offset{ coords.x - dpi->x, coords.y - dpi->y };
 
     // Check to make sure point is in the y range
-    if (y < 0)
+    if (offset.y < 0)
         return;
-    if (y >= dpi->height)
+    if (offset.y >= dpi->height)
         return;
     // Check to make sure we are drawing at least a pixel
     if (!no_pixels)
         no_pixels++;
 
-    x -= dpi->x;
-
     // If x coord outside range leave
-    if (x < 0)
+    if (offset.x < 0)
     {
         // Unless the number of pixels is enough to be in range
-        no_pixels += x;
+        no_pixels += offset.x;
         if (no_pixels <= 0)
             return;
         // Resets starting point to 0 as we don't draw outside the range
-        x = 0;
+        offset.x = 0;
     }
 
     // Ensure that the end point of the line is within range
-    if (x + no_pixels - dpi->width > 0)
+    if (offset.x + no_pixels - dpi->width > 0)
     {
         // If the end point has any pixels outside range
         // cut them off. If there are now no pixels return.
-        no_pixels -= x + no_pixels - dpi->width;
+        no_pixels -= offset.x + no_pixels - dpi->width;
         if (no_pixels <= 0)
             return;
     }
 
     // Get the buffer we are drawing to and move to the first coordinate.
-    uint8_t* bits_pointer = dpi->bits + y * (dpi->pitch + dpi->width) + x;
+    uint8_t* bits_pointer = dpi->bits
+        + offset.y * (static_cast<int64_t>(static_cast<int64_t>(dpi->pitch) + static_cast<int64_t>(dpi->width))) + offset.x;
```

Comment:
Why int64_t?

Reply:
![image](https://user-images.githubusercontent.com/12629817/117539649-98211100-afd9-11eb-8ef4-f98558235331.png)

Reply:
```suggestion
        + offset.y * (static_cast<int64_t>(dpi->pitch + dpi->width)) + offset.x;
```
I think this should be enough

Reply:
![image](https://user-images.githubusercontent.com/12629817/117543779-4afa6a80-afec-11eb-8815-1457597f7a03.png)

Reply:
Ok we can mark this one resolved as is.